stdlib/public/Concurrency/ExecutorImpl.h

```diff
@@ -118,7 +111,11 @@ static inline SwiftJobPriority swift_job_getPriority(SwiftJob *job) {
   return (SwiftJobPriority)((job->flags >> 8) & 0xff);
 }
 
-/// Allocate memory associated with a job.
+/// Allocate memory associated with a job.  Memory is allocated using a
+/// per-task allocator and will be disposed of automatically when the
+/// task is cleaned up.
+///
+/// N.B. **Requires that the job kind is SwiftJobKindTask.**
 void *swift_job_alloc(SwiftJob *job, size_t size);
```

Comment:
There is also a "stack discipline" requirement when using swift_job_alloc+swift_job_dealloc, correct? If yes, sound like it would be good to document in a comment, too.

Reply:
Good point, we should note that fact.